### DISCUSSION TITLE
Clear inputs on F4 stop

### DISCUSSION THREAD

**Discussion by ericrtew** [nirgoren/Eddienput#13] [Ideas]
Not sure if this is a bug with 02um or a feature not yet implemented, but currently I have to alt-tab after hitting F4 whatever my last inputs are stay held, and when I hit F3 after this it starts from those held inputs and breaks the first commands I had ready. Attached is a link to a video of me running a script, pausing during it, then rerunning it, and lastly dropping focus which then clears the up input. Seems like F4 should default to clearing this.

https://streamable.com/glie4u

**Comment by ericrtew** [comment]
Also this happens if you don't clear [] commands with a ][ even if you don't hit F4. For example a script that ends in [D] but no ]D[ on replays will continue to hold D. Is that intended functionality?

**Comment by nirgoren** [comment]
Pressing F4 is supposed to stop the recording and release all currently held buttons. The second part currently doesn't apply to buttons that are mapped to keyboard keys (like configs\kof_keyboard.json).

As for not clearing held buttons when the recording ends (naturally) - this is an intended behaviour. Please make sure to release them by the end of the recording.
